src/main/java/org/jetbrains/appenders/JsonLayout.java

```diff
@@ -33,137 +33,124 @@
-        fieldLabels.put("mdc", "mdc");
-        fieldLabels.put("ndc", "ndc");
-        fieldLabels.put("host", "host");
-        fieldLabels.put("path", "path");
-        fieldLabels.put("tags", "tags");
-        fieldLabels.put("@timestamp", "@timestamp");
-        fieldLabels.put("thread", "thread");
-        fieldLabels.put("@version", "@version");
-    }
-
-    private void updateFieldLabel(String key, String value) {
-        fieldLabels.put(key, value);
-    }
-
-    private String getFieldLabel(String key) {
-        return fieldLabels.get(key);
-    }
+    private class LoggerField {
+        private String defaultLabel;
+        private String renderedLabel;
+        private boolean isEnabled = true;
 
-    private String getLocationFieldLabel(String key) {
-        return getFieldLabel("location." + key);
-    }
+        LoggerField(String defaultName) {
+            this.defaultLabel = defaultName;
+            this.renderedLabel = defaultName;
+        }
 
-    private String getExceptionFieldLabel(String key) {
-        return getFieldLabel("exception." + key);
-    }
+        LoggerField(String defaultName, String renderedLabel) {
+            this.defaultLabel = defaultName;
+            this.renderedLabel = renderedLabel;
+        }
 
-    private enum LocationField {
-        CLASS("class"),
-        FILE("file"),
-        LINE("line"),
-        METHOD("method");
+        void updateRenderedLabel(String label) {
+            renderedLabel = label;
+        }
 
-        private final String val;
+        void disable() {
+            isEnabled = false;
+        }
 
-        LocationField(String val) {
-            this.val = val;
+        void enable() {
+            isEnabled = true;
         }
     }
 
-    private enum ExceptionField {
-        CLASS("class"),
-        MESSAGE("message"),
-        STACKTRACE("stacktrace");
-
-        private final String val;
-
-        ExceptionField(String val) {
-            this.val = val;
+    private class RenderedFieldLabels {
+        final LoggerField locationClass;
+        final LoggerField locationFile;
+        final LoggerField locationLine;
+        final LoggerField locationMethod;
+
+        final LoggerField exceptionClass;
+        final LoggerField exceptionMessage;
+        final LoggerField exceptionStacktrace;
+
+        final LoggerField exception;
+        final LoggerField level;
+        final LoggerField location;
+        final LoggerField logger;
+        final LoggerField message;
+        final LoggerField mdc;
+        final LoggerField ndc;
+        final LoggerField host;
+        final LoggerField path;
+        final LoggerField tags;
+        final LoggerField timestamp;
+        final LoggerField thread;
+        final LoggerField version;
+
+        private final ArrayList<LoggerField> allFields = new ArrayList<LoggerField>();
+
+        RenderedFieldLabels() {
+            locationClass = new LoggerField("location.class", "class");
```

Comment:
you may join declaration and initialization. It will make code 2x shorted and 2x more readable